dandi/tests/test_upload.py

```diff
@@ -181,9 +181,20 @@ def test_upload_sync_folder(
         text_dandiset.dandiset.get_asset_by_path("subdir2/banana.txt")
 
 
-def test_upload_bids(mocker: MockerFixture, bids_dandiset: SampleDandiset) -> None:
+def test_upload_bids_invalid(
+    mocker: MockerFixture, bids_dandiset_invalid: SampleDandiset
+) -> None:
     iter_upload_spy = mocker.spy(LocalFileAsset, "iter_upload")
-    bids_dandiset.upload(existing="forced")
+    with pytest.raises(RuntimeError):
+        bids_dandiset_invalid.upload(existing="forced")
+    iter_upload_spy.assert_not_called()
```

Comment:
Per our [discussion](https://github.com/dandi/dandi-cli/pull/1080#discussion_r924875265) about exception, should this be working:

```suggestion
    iter_upload_spy.assert_not_called()
    bids_dandiset_invalid.upload(existing="forced", validation="ignore")
    iter_upload_spy.assert_called()
```
? if so let's add it in?

Reply:
sounds good, thansk